src/Uno.Extensions.Navigation.UI/NavigationHostedService.cs

```diff
@@ -1,12 +1,17 @@
 ﻿namespace Uno.Extensions.Navigation.UI;
 
-internal record NavigationHostedService(ILogger<NavigationRegion> RegionLogger) : IHostedService
+internal record NavigationHostedService(ILogger<NavigationRegion> RegionLogger) : IHostedService, IStartupService
 {
+	private TaskCompletionSource<object> _completion = new TaskCompletionSource<object>();
+
 	public Task StartAsync(CancellationToken cancellationToken)
 	{
 		Region.Logger = RegionLogger;
+		_completion.SetResult(true);
```

Comment:
I don't think the generic version of TCS is needed

```suggestion
	private TaskCompletionSource _completion = new TaskCompletionSource();

	public Task StartAsync(CancellationToken cancellationToken)
	{
		Region.Logger = RegionLogger;
		_completion.SetResult();
```

Reply:
Doesn't exist for uap